src/value/number.rs

```diff
@@ -102,50 +102,69 @@ impl Zero for Number {
 
 impl<'a> fmt::Display for Formatted<'a, Number> {
     fn fmt(&self, out: &mut fmt::Formatter) -> fmt::Result {
-        let t = self.value.value.to_integer();
-        let skip_zero = self.format.is_compressed() || !self.value.lead_zero;
-        if t == 0 {
-            if self.value.value.is_negative() {
-                out.write_str("-")?;
-                if !skip_zero {
-                    out.write_str("0")?;
+        if self.value.value.is_negative() {
+            out.write_char('-')?;
+        }
+
+        let mut whole = self.value.value.to_integer().abs();
+        let mut dec = String::new();
```

Comment:
Ah.  I wanted to avoid extra allocation for a temporary string here.  But with it, you have implemented proper backtracking, which I have failed to implement without it.  So I guess it's worth it.  :sunglasses:

Reply:
I switched to using `String::with_capacity()` [here](https://github.com/kaj/rsass/pull/67/commits/0ec25dd878c34b40dfd5a7b2c3ccf25de75312f6#diff-1ce7630cffbee60699ecf546bda04efcL110), though I've no idea if it makes a difference in this case